contracts/crowdsale/ZxcCrowdsale.sol

```diff
@@ -0,0 +1,278 @@
+pragma solidity ^0.4.24;
+
+import "@0xcert/ethereum-utils/contracts/math/SafeMath.sol";
+import "@0xcert/ethereum-utils/contracts/ownership/Ownable.sol";
+import "@0xcert/ethereum-zxc/contracts/tokens/Zxc.sol";
+
+
+/**
+ * @title ZXC crowdsale contract.
+ * @dev Crowdsale contract for distributing ZXC tokens.
+ */
+contract ZxcCrowdsale is Ownable {
+  using SafeMath for uint256;
+
+  /**
+   * Token being sold.
+   */
+  Zxc public token;
+
+  /**
+   * Start timestamps for the following stages (start dates are inclusive, end exclusive):
+   * - Token pre-sale with 10% bonus: 2018/06/26 - 2018/07/04
+   * - Token sale with 5% bonus: 2018/07/04 - 2018/07/05
+   * - Token sale with 0% bonus: 2018/07/05 - 2018/07/18
+   */
+  uint256 public startTimePresale;
+  uint256 public startTimeSaleWithBonus;
+  uint256 public startTimeSaleNoBonus;
+
+  uint256 public bonusPresale;
+  uint256 public bonusSale;
+
+  /**
+   * End timestamp to end the crowdsale.
+   */
+  uint256 public endTime;
+
+  /**
+   * Minimum required wei deposit.
+   */
+  uint256 public minimumWeiDeposit;
+
+  /**
+   * Crowdsale cap in ZXC.
+   */
+  uint256 public crowdSaleZxcCap;
+
+  /**
+   * Amount of ZXC tokens sold.
+   */
+  uint256 public zxcSold;
+
+  /**
+   * Address where funds are collected.
+   */
+  address public wallet;
+
+  /**
+   * How many token units buyer gets per wei.
+   */
+  uint256 public rate;
+
+  /**
+   * @dev An event which is triggered when tokens are bought.
+   * @param _from The address sending tokens.
+   * @param _to The address recieving tokens.
+   * @param _weiAmount Purchase amount in wei.
+   * @param _tokenAmount The amount of purchased tokens.
+   */
+  event TokenPurchase(
+    address indexed _from,
+    address indexed _to,
+    uint256 _weiAmount,
+    uint256 _tokenAmount
+  );
+
+  /**
+   * @dev Contract constructor.
+   * @param _walletAddress Address of the wallet which collects funds.
+   * @param _tokenAddress Address of the ZXC token contract.
+   * @param _startTimePresale Start time of presale stage.
+   * @param _startTimeSaleWithBonus Start time of public sale stage with bonus.
+   * @param _startTimeSaleNoBonus Start time of public sale stage with no bonus.
+   * @param _endTime Time when sale ends.
+   * @param _rate ZXC/ETH exchange rate.
+   * @param _crowdSaleZxcCap Number of ZXC tokens offered during the sale.
+   * @param _bonusPresale Bonus token percentage for presale.
+   * @param _bonusSale Bonus token percentage for public sale stage with bonus.
+   * @param _minimumWeiDeposit Minimum required deposit in wei.
+   */
+  constructor(address _walletAddress,
```

Comment:
Change the code style of input parameters to align with solidity code style (so the same way TokenPurchase event is).

Reply:
1426e15